### PR TITLE
test: move TestProfiling to standalone APM Server

### PR DESCRIPTION
**Pull request by kruskall**
## Motivation/summary

Reduce fleet dependency in test and improve test time.
TestProfiling time went from 32s to 10s

## Checklist

- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [ ] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

### CODE REVIEW

File: systemtest/profiling_test.go [hunk 1]
```diff
@@ -82,31 +83,20 @@ func TestProfiling(t *testing.T) {
...
-							"hosts":   []string{"elasticsearch:9200"},
-							"api_key": apiKey.ID + ":" + apiKey.APIKey,
-						},
-					},
-				},
-			},
-		})
-
-	apmServerURL, _ := url.Parse(apmIntegration.URL)
+	srv := apmservertest.NewUnstartedServerTB(t)
```
Comment (axw) [inline systemtest/profiling_test.go:86]: With this change, the comment at line 58 doesn't make sense, and the test doesn't quite cover what it was doing. We could simulate what Elastic Agent does by creating a more limited API Key for the Elasticsearch output.

Comment (kruskall) [reply systemtest/profiling_test.go:86]: Good catch! Wouldn't the behaviour be the same with standalone ? In this case: the standalone APM Server is using the default elasticsearch credentials (with narrow privileges) and we are creating and passing an api_key to write to the profiling indices.

Comment (axw) [reply systemtest/profiling_test.go:86]: Ah, true. I was thinking the standalone apm-server run by system tests used superuser privileges, but they don't. So, I think we just need to update the comment then.

Comment (kruskall) [reply systemtest/profiling_test.go:86]: Updated! :+1:

File: systemtest/apmservertest/config.go [hunk 2]
```diff
@@ -69,6 +69,9 @@ type Config struct {
 
 	// Output holds configuration for the libbeat output.
 	Output OutputConfig `json:"output"`
+
+	// ProfilingConfig holds configuration related to profiling.
+	Profiling *ProfilingConfig `json:"apm-server.profiling"`
```
Comment (axw) [inline systemtest/apmservertest/config.go:74]:
```suggestion
	Profiling *ProfilingConfig `json:"apm-server.profiling,omitempty"`
```

File: systemtest/apmservertest/config.go [hunk 3]
```diff
@@ -169,6 +172,28 @@ type TailSamplingPolicy struct {
...
+
+	// ILMConfig
+	ILMConfig *ProfilingILMConfig `json:"keyvalue_retention"`
+}
+
+type ProfilingILMConfig struct {
+	Age         time.Duration `json:"age"`
+	SizeInBytes uint64        `json:"size_bytes"`
+	Interval    time.Duration `json:"execution_interval"`
+}
```
Comment (axw) [inline systemtest/apmservertest/config.go:195]: I think we should leave ILMConfig out, and remove the ProfilingILMConfig. It's not used AFAICS. If it were, I think we'd need to add a custom MarshalJSON method like in other structs, for marshalling the time.Duration as expected.

**Review by axw — APPROVED** [review]
LGTM with one minor change.

File: systemtest/apmservertest/config.go (hunk 3 above)
Comment (kruskall) [reply systemtest/apmservertest/config.go]: Good catch! Updated!

**Pull request merged**